client/state/site-keyrings/reducer.js

```diff
@@ -0,0 +1,93 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { keyBy, omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer, keyedReducer } from 'state/utils';
+import { siteKeyrings as siteKeyringsSchema } from './schema';
+import {
+	SITE_KEYRINGS_REQUEST,
+	SITE_KEYRINGS_REQUEST_FAILURE,
+	SITE_KEYRINGS_REQUEST_SUCCESS,
+	SITE_KEYRINGS_SAVE,
+	SITE_KEYRINGS_SAVE_FAILURE,
+	SITE_KEYRINGS_SAVE_SUCCESS,
+	SITE_KEYRINGS_DELETE_SUCCESS,
+} from 'state/action-types';
+
+/**
+ * Returns the updated requests state after an action has been dispatched. The
+ * state maps site ID to whether a request is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const requesting = createReducer(
+	{},
+	{
+		[ SITE_KEYRINGS_REQUEST ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: true } ),
+		[ SITE_KEYRINGS_REQUEST_SUCCESS ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+		[ SITE_KEYRINGS_REQUEST_FAILURE ]: ( state, { siteId } ) => ( { ...state, [ siteId ]: false } ),
+	}
+);
+
+/**
+ * Returns the save Request status after an action has been dispatched. The
+ * state maps site ID to the request status
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const saveRequests = createReducer(
+	{},
+	{
+		[ SITE_KEYRINGS_SAVE ]: ( state, { siteId } ) => ( {
+			...state,
+			[ siteId ]: { saving: true, status: 'pending', error: false },
+		} ),
+		[ SITE_KEYRINGS_SAVE_SUCCESS ]: ( state, { siteId } ) => ( {
+			...state,
+			[ siteId ]: { saving: false, status: 'success', error: false },
+		} ),
+		[ SITE_KEYRINGS_SAVE_FAILURE ]: ( state, { siteId, error } ) => ( {
+			...state,
+			[ siteId ]: { saving: false, status: 'error', error },
+		} ),
+	}
+);
+
+const siteKeyrings = createReducer(
+	{},
+	{
+		[ SITE_KEYRINGS_REQUEST_SUCCESS ]: ( state, { keyrings } ) => keyBy( keyrings, 'keyring_id' ),
```

Comment:
We can't actually use `keyBy` here. There can be multiple items with the same keyring_id so we should keep keyrings as a list and filter on it